fl_training.py

```diff
@@ -205,13 +205,266 @@ def build_aggregated_model(new_weights):
+    is_very_first_minibatch = (epoch_index == 0 and minibatch_index == 0)
+    local_score_list, model_list, score_matrix, score_matrix_extended = iterative_results
+    minibatched_x_train, minibatched_y_train = train_data
+
+    # Starting model for each partner is the aggregated model from the previous mini-batch iteration
+    if is_very_first_minibatch:  # Except for the very first mini-batch where it is a new model
+        logger.debug(f"(fedavg) Very first minibatch of epoch n°{epoch_index}, init new models for each partner")
+        partners_model_list_for_iteration = init_with_new_models(partners_list)
+    else:
+        logger.debug(f"(fedavg) Minibatch n°{minibatch_index} of epoch n°{epoch_index}, "
+                     f"init aggregated model for each partner with models from previous round")
+        partners_model_list_for_iteration = init_with_agg_models(
+            partners_list, aggregation_weighting_approach, local_score_list, model_list)
+
+    # Iterate over partners for training each individual model
+    for partner_index, partner in enumerate(partners_list):
+
+        # Reference the partner's model
+        partner_model = partners_model_list_for_iteration[partner_index]
+
+        # Train on partner local data set
+        train_data_for_fit_iteration = (
+            minibatched_x_train[partner_index][minibatch_index],
+            minibatched_y_train[partner_index][minibatch_index],
+        )
+        history = collaborative_round_fit(partner_model, train_data_for_fit_iteration, val_data, partner.batch_size)
+
+        # Print results of the round
+        print_collaborative_round_partner_result(
+            partner, partner_index, partners_count, collaborative_round_indexes, history.history["val_accuracy"][0])
+
+        # Update the partner's model in the models' list
+        model_list[partner_index] = partner_model
+
+        # Update iterative results
+        update_iterative_results(partner_index, collaborative_round_indexes, history, iterative_results)
+
+    logger.debug("End of fedavg collaborative round.")
+
+
+def compute_collaborative_round_sequential(
+        partners_list,
+        sequentially_trained_model,
+        val_data,
+        iterative_results,
+        train_data,
+        collaborative_round_indexes,
+):
+    """Proceed to a collaborative round with a sequential approach"""
+
+    logger.debug("Start new sequential collaborative round ...")
+
+    # Initialize variables
+    partners_count = len(partners_list)
+    epoch_index, epoch_count, minibatch_index, minibatch_count = collaborative_round_indexes
+    minibatched_x_train, minibatched_y_train = train_data
+
+    # Iterate over partners for training the model sequentially
+    shuffled_indexes = np.random.permutation(len(partners_list))
+    logger.debug(f"(seq) Shuffled order for this sequential collaborative round: {shuffled_indexes}")
+    for for_loop_idx, partner_index in enumerate(shuffled_indexes):
```

Comment:
What is `for_loop_idx `??

Reply:
The index of iterations over the for loop